server/tests/integration/test_api_group.py

```diff
@@ -21,153 +21,166 @@
-		self.user = self.accounts['dummy_student']
-		inst = self.get_basic_instance()
-		inst.invited.append(self.user.key)
-		inst.put()
-
-		self.post_json('/{}/{}/decline'.format(self.name, inst.key.id()))
-
-		self.assertEqual(inst.invited, [])
-		self.assertNotIn(self.user.key, inst.member)
-
-	def test_exit_member(self):
-		self.user = self.accounts['dummy_student']
-		inst = self.get_basic_instance()
-		inst.member.append(self.user.key)
-		inst.put()
-
-		self.post_json(
-			'/{}/{}/remove_member'.format(self.name, inst.key.id()),
-			data={'email': self.user.email[0]})
-
-		self.assertNotIn(self.user.key, inst.member)
-
-	def test_invite_someone_in_a_group(self):
-		self.user = self.accounts['dummy_student2']
-		inst = self.get_basic_instance()
-		inst.put()
-
-		# Place dummy_student in another group
-		invited = self.accounts['dummy_student']
-		self.model(
-			assignment=self.assignment.key,
-			member=[invited.key, self.accounts['dummy_staff'].key]
-		).put()
-
-		self.post_json(
-			'/{}/{}/add_member'.format(self.name, inst.key.id()),
-			data={'email': invited.email[0]})
-
-		self.assertStatusCode(400)
-		self.assertEqual(inst.invited, [])
-
-	def test_remove_from_two_member_group(self):
-		self.user = self.accounts['dummy_student']
-		inst = self.model(assignment=self.assignment.key,
-		                  member=[self.user.key, self.accounts['dummy_student2'].key])
-		inst.put()
-
-		self.post_json(
-			'/{}/{}/remove_member'.format(self.name, inst.key.id()),
-			data={'email': self.user.email[0]})
+  model = models.Group
+  name = 'group'
+  num = 1
+  access_token = 'dummy_admin'
+
+  def setUp(self):
+    super(GroupAPITest, self).setUp()
+    self.course = make_fake_course(self.user)
+    self.course.put()
+    self.assignment = make_fake_assignment(self.course, self.user)
+    self.assignment.put()
+    for student_name in [a for a in self.accounts if 'student' in a]:
+      s = self.accounts[student_name]
+      models.Participant.add_role(s, self.course, constants.STUDENT_ROLE)
+
+  def get_basic_instance(self, mutate=True):
+    return self.model(
+      assignment=self.assignment.key,
+      member=[
+        self.accounts['dummy_student2'].key,
+        self.accounts['dummy_student3'].key])
+
+  def test_assignment_group(self):
+    self.user = self.accounts['dummy_student2']
+    inst = self.get_basic_instance()
+    inst.put()
+
+    self.get('/assignment/{}/group'.format(self.assignment.key.id()))
+    self.assertEqual(self.response_json['id'], inst.key.id())
+
+  def test_index_arguments(self):
+    self.user = self.accounts['dummy_student2']
+    self.partner = self.accounts['dummy_student3']
+    inst = self.get_basic_instance()
+    inst.put()
+
+    self.get_index(assignment=self.assignment.key.id(), member=self.user.key.id())
+    self.assertEqual(self.response_json[0]['id'], inst.key.id())
+
+    self.get_index(member=self.partner.key.id())
+    self.assertEqual(self.response_json[0]['id'], inst.key.id())
```

Comment:
The above function (`test_index_arguments`) is the actual addition to this file.